docs/storage.md

```diff
@@ -449,6 +450,112 @@ config:
   directory: ""
 ```
 
+### OCI
+
+[Oracle Cloud Infrastructure](https://www.oracle.com/cloud). To configure OCI Object Storage as Thanos Object Store, you need access to OCI Object Storage.
+You can get a free OCI Cloud Services by following the [Oracle Cloud Free Tier](https://www.oracle.com/cloud/free) page.
+More information about [Overview of Object Storage](https://docs.cloud.oracle.com/en-us/iaas/Content/Object/Concepts/objectstorageoverview.htm).
+
+OCI Object Storage supports three different providers, default, instance principle, and raw. 
+The default HTTP Client comes with this Thanos OCI Object Storage should be able to handle most of the use cases.
+
+By default, like other clients, Thanos OCI Object Storage client will storage the data from a single prometheus instance to the top directory of the bucket. 
+This means each bucket can only store one prometheus instance data.
+
+To support multiple prometheus instances with a single bucket, a new `object_base_path` option is introduced in OCI Object Storage client. 
+If defined, the value will be used as the prefix of the prometheus data in OCI bucket.
+
+For example, if you want to use the same bucket for 2 prometheus instances, you can define `object_base_path` as `/path/for/prometheus1` 
+for the first prometheus instance, and `/path/for/prometheus2` for the second prometheus instance.
+
+#### Default Provider
+The default config provider will look for configurations in 3 places: file in `$HOME/.oci/config`, `$HOME/.obmcs/config`, and
+variables names starting with the string TF_VAR. If the same configuration is found in multiple places the provider will
+prefer the first one.
+
+For Example:
+```yaml
+type: OCI
+config:
+  provider: "default"
+  bucket: ""
+  compartment_ocid: ""
+  object_base_path: ""            // Optional object base path (prefix) in OCI bucket.
```

Comment:
Please see the explanation above for  _**What does this mean? Why only one?**_
Hope it will make more sense after you read my explanation.

If you want to use different variable name or an alternative approach for this new objectstorage prefix feature, please let me know.